api/system.cc

```diff
@@ -147,21 +147,17 @@ void set_system(http_context& ctx, routes& r) {
 
     hs::set_io_limit.set(r, [&ctx](std::unique_ptr<request> req) {
         try {
-            return ctx.db.invoke_on_all([&req] (replica::database& db) {
-                auto io_class = req.get_path_param("class_name");
-                auto bandwidth = boost::lexical_cast<uint64_t>(req.get_query_param("bandwidth"));
-                auto iops = boost::lexical_cast<uint64_t>(req.get_query_param("iops"));
-
-                return db.set_io_limits(io_class, bandwidth, iops).get();
-            }).then([] {
-                apilog.info("IO limits updated successfully");
-                return json::json_return_type(json::json_void());
-            });
+            auto io_class = req->get_path_param("class_name");
+            auto bandwidth = boost::lexical_cast<uint64_t>(req->get_query_param("bandwidth"));
+            auto iops = boost::lexical_cast<uint64_t>(req->get_query_param("iops"));
+
+            ctx.db.local().set_io_limits(io_class, bandwidth, iops).get();
```

Comment:
Yikes! This code is not running in seastar::async() AFAIK. Either co_await or return the future

Reply:
@xemul fixed